File: src/main/java/xyz/acrylicstyle/plugin/PluginManagerConfig.java
package xyz.acrylicstyle.plugin;

import org.bukkit.ChatColor;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;
import xyz.acrylicstyle.tomeito_api.providers.ConfigProvider;
import xyz.acrylicstyle.tomeito_api.providers.LanguageProvider;

public class PluginManagerConfig extends ConfigProvider {
    public PluginManagerConfig(String path) {
        super(path);
    }

    @NotNull
    @Contract("_ -> new")
    public static PluginManagerConfig newInstance(@NotNull String path) {
        return new PluginManagerConfig(path);
    }

    public String getLanguage() {
        return this.getString("language", "en_US");
    }

    public static LanguageProvider getCurrentLanguageStatic() {
        return PluginManager.config.getCurrentLanguage();
    }

    public static String getStringStatic(String key) {
        return ChatColor.translateAlternateColorCodes('&', getCurrentLanguageStatic().getString(key));
    }

    public static String getStringStatic(String key, String def) {
        return ChatColor.translateAlternateColorCodes('&', getCurrentLanguageStatic().getString(key, def));
    }

    public LanguageProvider getCurrentLanguage() {
        return PluginManager.lang.get(getLanguage());
    }

    public void setLanguage(String language) {
        this.set("language", language);
    }
}

File: src/main/java/xyz/acrylicstyle/plugin/PluginManagerConfig.kt
package xyz.acrylicstyle.plugin

import org.bukkit.ChatColor
import org.jetbrains.annotations.Contract
import xyz.acrylicstyle.plugin.utils.ConfigProvider
import xyz.acrylicstyle.plugin.utils.LanguageProvider

class PluginManagerConfig(path: String?) : ConfigProvider(path!!) {
    var language: String
        get() = this.getString("language", "en_US")
        set(language) {
            this["language"] = language
        }
    val currentLanguage: LanguageProvider
        get() = PluginManager.lang[language]

    companion object {
        @Contract("_ -> new")
        fun newInstance(path: String): PluginManagerConfig {
            return PluginManagerConfig(path)
        }

        private val currentLanguageStatic: LanguageProvider
            get() = PluginManager.config.currentLanguage

        fun getString(key: String?): String {
            return ChatColor.translateAlternateColorCodes('&', currentLanguageStatic.getString(key))
        }

        fun getString(key: String?, def: String?): String {
            return ChatColor.translateAlternateColorCodes('&', currentLanguageStatic.getString(key, def))
        }
    }
}